src/main/java/kafdrop/config/OASConfiguration.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017 Kafdrop contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package kafdrop.config;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ *  Autoconfiguration for OpenAPI Specification (OAS).
+ *  Can be disabled by setting {@code springdoc.api-docs.enabled=false}.
+ */
+@Configuration
+@ConditionalOnProperty(value = "springdoc.api-docs.enabled", matchIfMissing = true)
+public class OASConfiguration {
+
+  @Bean
+  public OpenAPI httpApi() {
```

Comment:
Thanks for the rename!
This condition you don't need. SpringDoc already implements this behavior, see [here](https://springdoc.org/#disabling-the-springdoc-openapi-endpoints).